gateway/cert.go

```diff
@@ -392,6 +397,11 @@ func (gw *Gateway) getTLSConfigForClient(baseConfig *tls.Config, listenPort int)
 		}
 
 		for _, spec := range gw.apiSpecs {
+			// eliminate APIs which are not in the current port
+			if (spec.ListenPort == 0 && controlAPIPort != listenPort) || (spec.ListenPort != 0 && spec.ListenPort != listenPort) {
```

Comment:
Can you please split this into two if statements, with a comment explaining each condition, something along:

// skip the control api port
// skip the main listen port